### PR TITLE
Establish rust project layout

### PR DESCRIPTION
**Pull request by Xuanwo**
This PR will establish rust project layout.

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
also add a notice file? copyright file?

**Comment by Xuanwo** [comment]
> also add a notice file? copyright file?

I'm going to add asf related stuffs in following PRs.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Contributor Covenant Code of Conduct
```
[the remark this answers is not in the document]
Comment (Xuanwo) [reply CODE_OF_CONDUCT.md]: DONE.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**